core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -341,7 +341,9 @@ private class LiveExecutorStageSummary(
       metrics.shuffleWriteMetrics.recordsWritten,
       metrics.memoryBytesSpilled,
       metrics.diskBytesSpilled,
-      isBlacklisted)
+      isBlacklisted,
```

Comment:
these are actually going to be written to the tracking store.  we don't want that if the values aren't set.

Reply:
@vanzin ideas on how to better handle this?

I don't see a real clean way to populate these fields from the AppstatusListener before being written.    For context, in this PR these are currently being populated in the AppStatusStore.executorSummary call before going back to user.

We could potentially split into separate api or on UI side query both that and the executor info and join but seems like a lot more data.

Reply:
I don't think the executor stage summary should contain this info.

Is it really that difficult to do the join? The current stage page already does that:
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala#L710

Is the concern that doing something like that from JS would be expensive or too difficult?

Reply:
We can look into it more, I can't image its that hard to join, its just that much more data that is going over the wire, you are sending the entire ExecutorSummary for all executors when you really just need 2 fields out of it of some executors.  Previously this was happening on the driver side so it didn't have to transfer the data.  We can look at the data size and if it doesn't seem to bad we can do that for now and can always revisit if needed.  The stage page executor table which uses this I wouldn't expect to be as used as the task one so probably not bad for now.

Reply:
> you are sending the entire ExecutorSummary for all executors when you really just need 2 fields out of it of some executors

The current scala code avoids that by just fetching executor summaries it needs; the problem I can see from JS is that it would cause potentially many requests to the driver (which are more expensive than a hash table or level db lookup), but maybe it's not so bad if it's restricted to what is being shown in a single page.

Reply:
Have made the required changes and tested them.